spyder/plugins/toolbar/plugin.py

```diff
@@ -71,6 +72,17 @@ def on_main_menu_available(self):
             section=ViewMenuSections.Toolbar,
             before_section=ViewMenuSections.Layout)
 
+    @on_plugin_teardown(plugin=Plugins.MainMenu)
+    def on_main_menu_teardown(self):
+        mainmenu = self.get_plugin(Plugins.MainMenu)
+        # View menu Toolbar section
+        mainmenu.remove_item_from_application_menu(
+            'toolbars_menu',
+            menu_id=ApplicationMenus.View)
+        mainmenu.remove_item_from_application_menu(
+            'show toolbars',
+            menu_id=ApplicationMenus.View)
+

```

Comment:
This also requires to change its `on_close` method to be like this:

```python
def on_close(self):
    container = self.get_container()
    container._save_visible_toolbars()
    for toolbar in container._visible_toolbars:
        toolbar.setVisible(False)
```

Also, the method `load_last_visible_toolbars` in the container is missing the `for` loop showm below:

```python
def load_last_visible_toolbars(self):
    """Load the last visible toolbars from our preferences.."""
    toolbars_names = self.get_conf('last_visible_toolbars')

    if toolbars_names:
        toolbars_dict = {}
        for toolbar in self._toolbarslist:
            toolbars_dict[toolbar.objectName()] = toolbar

        toolbars = []
        for name in toolbars_names:
            if name in toolbars_dict:
                toolbars.append(toolbars_dict[name])

        self._visible_toolbars = toolbars
    else:
        self._get_visible_toolbars()

    # --- THIS IS MISSING!! ----
    for toolbar in self._visible_toolbars:
        toolbar.setVisible(True)

    self.update_actions()
```